toolchain/check/testdata/facet/facet_assoc_const.carbon

```diff
@@ -206,6 +206,20 @@ fn G(T:! M where .X = () and .Y = ()) {
   F(T);
 }
 
+// --- repeated_concrete_value_and_associated.carbon
+library "[[@TEST_NAME]]";
+
+interface M { let X:! type; let Y:! type; }
+
+fn F1(T:! M where .X = () and .Y = .X and .X = .Y) {}
+
+fn F2(T:! M where .X = () and .X = .X) {}
+
+fn G(T:! M where .X = () and .Y = ()) {
+  F1(T);
+  F2(T);
+}
+
```

Comment:
Do we have a test where the rewrites are given in the opposite order and it fails with a cycle error?